### DISCUSSION TITLE
trial table creation problem

### DISCUSSION THREAD

**Discussion by dylann916** [BioMotionLab/TUX#78] [Q&A]
Hello,
I'm currently running a psychology experiment using your module. In this one I have 1 block (independant) variable with two modalities and an independent variable with two modalities (Go/NoGo) whose proportions must be 2/3 Go trials and 1/3 NoGo trials. Finally, I have a 3-modality "TargetDepth" variable, which must have equal probabilities. What I want is to have a tree structure like this: with 648 trials in total, for example, I'll have 324 trials in one modality of my "difficulty load" block variable and 324 in the other modality. Then within each modality I'd have 2/3 Go trials, i.e. 216 trials, and 1/3 NoGo trials, i.e. 108 trials. Then I'd like to have as many trials in each type of trial (Go/NoGo) in each modality of the "TargetDepth" variable, i.e. 72 trials in each condition. The example here isn't very important, but what is important is that when I declare these variables (see screen) I get a completely buggy table of trials, and I don't understand why. For example, on the screen I sent you of 1) the Go/NoGo proportions are not respected and it seems that certain modalities of the "TargetDepth" variable are systematically associated with certain modalities of the "GoNoGo" variable. For example, here, the NoGo trials are systematically associated with the "Near" modality and the Go trials are systematically associated with the "Middle" and "Far" modalities. Sometimes, too, the table will display only Go trials. It seems that the module doesn't support more than three variables, and doesn't seem to support changes in the probability of certain variables... However, I've already done experiments with 2 variables and everything worked perfectly. Do you know where this is coming from and how to fix the problem?

<img width="281" alt="Image1" src="https://github.com/BioMotionLab/TUX/assets/118400259/27335dc3-f0ef-4aaf-92f9-dec8068f3196">
<img width="276" alt="Image2" src="https://github.com/BioMotionLab/TUX/assets/118400259/0135e206-f904-43f4-b6b2-723daa7d5175">
<img width="290" alt="Image3" src="https://github.com/BioMotionLab/TUX/assets/118400259/bd3d0918-9811-4695-8e3c-140049d4242f">
<img width="533" alt="Screenshot 2024-04-12 223038" src="https://github.com/BioMotionLab/TUX/assets/118400259/fd6cf623-41ca-4398-bcc1-665b1113d3b8">

**Comment by AdamBebko** [comment]
You you please provide a bit more info? your entire experiment config file screenshot, and the entire trial table please? I'll try to look into this in the next few days.
